### ISSUE TITLE
Distraction free mode: Keyboard shortcut not working 

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#47873]
### Description

Related to: #41740

Pressing the shortcut key indicated in the options menu does not switch to the distraction-free mode.

Shortcuts to switch to full-screen mode or to the code editor both work correctly. I tested on Windows, but would be happy if someone could test to see if this problem reproduces on a Mac as well.

### Step-by-step reproduction instructions

- Open the editor.
- Press `Ctrl` + `Shift` + `\` key (This is a shortcut for Windows) .
- Nothing happens

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/54422211/217525824-aa96dc73-d2b9-44c0-81eb-8d60d9906b53.mp4

### Environment info

- OS: Windows 11
- Wordpress Version: WordPress 6.2-alpha-55250
- Gutenberg Version: Version 15.1.0-rc.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by skorasaurus** [comment]
I'm able to reproduce in ubuntu 18.04; Gutenberg 15.0.1; WordPress 6.1; nothing happens with the Ctrl + Shift + \ key.

**Comment by glendaviesnz** [comment]
On a Mac the `shift - command - \`  (⇧ ⌘ \ ) works to toggle distraction free mode for me.

**Comment by t-hamano** [comment]
Thank you for the test!

I think the problem may be related to the keyboard layout.

I used  [JavaScript Key Code Event Tool](https://www.toptal.com/developers/keycode) to get a event dump. My Japanese keyboard layout showed the following results.

Just press the `\` key:

```json
{
  "key": "\\",
  "keyCode": 226,
  "which": 226,
  "code": "IntlRo",
  "location": 0,
  "altKey": false,
  "ctrlKey": false,
  "metaKey": false,
  "shiftKey": false,
  "repeat": false
}
```

Hold down `ctrl` and `shift` and press `\` key:

```json
{
  "key": "_",
  "keyCode": 226,
  "which": 226,
  "code": "IntlRo",
  "location": 0,
  "altKey": false,
  "ctrlKey": true,
  "metaKey": false,
  "shiftKey": true,
  "repeat": false
}
```

You can confirm that the value of `event.key` has changed. This may possibly be due to the fact that keyboard events consider modifier keys, as in the fix made in #43428.

On my Japanese keyboard, if I press the `shift` key and `\` at the same time, `_` is typed. If I try and change the shortcut character to another character, the distraction free mode toggle works correctly. Perhaps on the Mac, the `\` key doesn't have a combination key when the `shift` key is pressed, so this problem may not be reproducible in some cases.

I will look into this a bit more to see what approach would be appropriate.

**Comment by t-hamano** [comment]
Okay, perhaps the problem could be that the `shift` key has a character variation when it is pressed.
The complication is that the variations are not matched by the keyboard layout.

Here is what I have found out  (**on Windows**):

| Key                   | Pressed with the `shift` key<br />on Japanese keyboard layout | Pressed with the `shift` key<br />on Japanese keyboard layout |
| --------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
| `,`                   | `<`                                                          | `<`                                                          |
| `.`                   | `>`                                                          | `>`                                                          |
| `/`                   | `?`                                                          | `?`                                                          |
| `\` (top position)    | `|`                                                          | `|`                                                          |
| `\` (bottom position) | `_`                                                          | N/A                                                          |
| `;`                   | `+`                                                          | `:`                                                          |
| `:`                   | `*`                                                          | N/A                                                          |
| `]`                   | `}`                                                          | `}`                                                          |
| `@`                   | `` ` ``                                                      | N/A                                                          |
| `[`                   | `{`                                                          | `{`                                                          |
| `1`                   | `!`                                                          | `!`                                                          |
| `2`                   | `"`                                                          | `@`                                                          |
| `3`                   | `#`                                                          | `#`                                                          |
| `4`                   | `$`                                                          | `$`                                                          |
| `5`                   | `%`                                                          | `%`                                                          |
| `6`                   | `&`                                                          | `^`                                                          |
| `7`                   | `'`                                                          | `&`                                                          |
| `8`                   | `(`                                                          | `*`                                                          |
| `9`                   | `)`                                                          | `(`                                                          |
| `0`                   | N/A                                                          | `)`                                                          |
| `-`                   | `=`                                                          | `_`                                                          |
| `^`                   | `~`                                                          | N/A                                                          |
| =                     | N/A                                                          | `+`                                                          |

### Windows Japanese Keyboard Layout

![jp](https://user-images.githubusercontent.com/54422211/217714443-e00a932f-91bd-4c6b-885d-c1c6e6f05351.png)

### Windows English Keyboard Layout

![en](https://user-images.githubusercontent.com/54422211/217714456-feb72bc0-fd06-4465-a2d1-6bba2a3e458e.png)
